data-acquisition/data-synthesizer/src/data_synthesizer/data_synthsizer.py

```diff
@@ -0,0 +1,159 @@
+import os
+import random
+import numpy as np
+from PIL import Image
+import json
+import time
+
+OBJ_SIZE_MAP = {
+    "s": 0.6,
+    "m": 0.75,
+    "l": 0.9
+}
+
+
+class DataSynthesizer:
+
+    def __init__(self, config_path, template_path, data_path):
+        self.config_path = config_path
+        self.data_path = data_path
+        self.template_path = template_path
+        with open(self.config_path + "/backgrounds.json") as bg_conf_file:
+            self.bg_conf = json.load(bg_conf_file)
+        self.bg_labels = list(self.bg_conf.keys())
+        with open(self.config_path + "/foregrounds.json") as fr_conf_file:
+            self.fr_conf = json.load(fr_conf_file)
+
+    def _get_new_image_size(
+            self, curr_obj_conf, obj_sizes_allowed, shelf_ht):
+
+        curr_fr_obj_size = np.random.choice(obj_sizes_allowed, 1)[0]
+        new_obj_ht = int(shelf_ht * OBJ_SIZE_MAP[curr_fr_obj_size])
+        new_obj_width = int(
+            (new_obj_ht/curr_obj_conf["height"])*curr_obj_conf["width"])
+
+        new_img_size = (new_obj_width, new_obj_ht)
+
+        return new_img_size
+
+    def _get_fr_and_conf(self, categories):
+
+        curr_fr_category = np.random.choice(categories, 1)[0]
+        curr_fr_category_objs = list(self.fr_conf[curr_fr_category].keys())
+        curr_fr_obj = np.random.choice(curr_fr_category_objs, 1)[0]
+        curr_obj_conf = self.fr_conf[curr_fr_category][curr_fr_obj]
+
+        curr_obj_file = (
+            self.template_path + "/foregrounds/{cat}/{obj}.png").format(
+                cat=curr_fr_category, obj=curr_fr_obj)
+
+        return curr_obj_file, curr_obj_conf
+
+    def _get_bg_and_conf(self):
+        curr_bg_label = np.random.choice(self.bg_labels, 1)[0]
```

Comment:
If we're using `bg` for `background`, wouldn't it be more consistent to use `fg` for `foreground` (as opposed to `fr`)?

Reply:
Haha. Yeah consistency is key I guess :p